components/script/dom/bindings/js.rs

```diff
@@ -391,6 +392,79 @@ impl<T: DomObject> HeapSizeOf for MutNullableJS<T> {
     }
 }
 
+/// A holder that allows to lazily initialize the value only once
+/// `JS<T>`, using OnceCell
+/// Essentially a `OnceCell<JS<T>>`.
+///
+/// This should only be used as a field in other DOM objects; see warning
+/// on `JS<T>`.
+#[must_root]
+pub struct OnceCellJS<T: DomObject> {
+    ptr: OnceCell<JS<T>>,
+}
+
+impl<T: DomObject> OnceCellJS<T> {
+    /// Create a new `OnceCellJS` with value
+    pub fn new(initial: Option<&T>) -> OnceCellJS<T> {
```

Comment:
which makes this function similar to `Default::default()` Instead, we can change the signature to `new_with_value` similar to mitochondria